### DISCUSSION TITLE
[Craft 5]: Should the "New entry" button be just disabled OR hidden when a Matrix field has reached the "Max Entries" limit?

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#14264] [General]
Not sure about this, that's why I'm posting as a discussion and not a suggestion:

The current behaviour in Craft 5 is that the "New entry" button is _disabled_ when a Matrix field has reached the "Max Entries" limit, _except_ when the view mode is set to `Inline-editable blocks`. In this last case it is completely _hidden_.

**Is it by design that there are two different behaviours?** 

**Should the button be hidden in any/all of the other cases?**

- I like the fact that it is hidden for `inline-blocks view`. This let's me use a matrix field as a kind of [static field group (with some limitations)](https://github.com/craftcms/cms/discussions/14015). Purely a CP presentational thing.
- In `card view` we can also use matrix fields in a similar manner but the visible button doesn't fit with the "static field group" concept (which is a hack without native support, but kinda works with inline blocks).
- In `element index view` I think it works better as it is. In this view it's obvious that you add and remove whole entries, and not just group fields for presentation purposes.

Anyway, that's my 2 cents. It's a minor point, but maybe worth thinking about.

**Comment by mmikkel** [comment]
That's strange, the button is not hidden on my end, just disabled (which IMO is the appropriate behaviour).

<img width="1165" alt="CleanShot 2024-02-01 at 13 50 26@2x" src="https://github.com/craftcms/cms/assets/298510/8fb0cdb5-fc6b-49b3-bb7f-b2595e88a02d">

**Reply by thupsi** [reply]
Hmm, are both `Min entries` and `Max entries` set to `1`?

**Reply by mmikkel** [reply]
Initially I only had `Max entries` set to `1`, but I get the same result with `Min entries` set to `1` as well.

<img width="473" alt="CleanShot 2024-02-01 at 14 17 03@2x" src="https://github.com/craftcms/cms/assets/298510/c7e5dc66-3a19-4a41-be28-be32f9ec17dc">

FTR, I think disabling and not hiding the button is the correct behaviour. But, strange that we're seeing different things.

**Reply by thupsi** [reply]
I think I figured it out. You have to:

1. Have `Min Entries` be equal to `Max Entries`
2. Have only one entry type allowed in the field settings.
3. Add the maximum blocks according to step 1.

Can you confirm?

<img width="582" alt="image" src="https://github.com/craftcms/cms/assets/19857599/91c6bd94-08c0-4bef-83cf-42fb7877dca4">

<img width="386" alt="image" src="https://github.com/craftcms/cms/assets/19857599/9baa0631-9cb4-44fc-9674-7e001733b36d">

**Reply by mmikkel** [reply]
Aha. Yeah, can confirm; when there's only one entry type as per your setup above, the button disappears once an entry is created in the field. That's odd – feels unintended to me but who knows.

**Reply by thupsi** [reply]
It is a little weird, I agree, and I'm not sure it's intended!

I also agree that strictly speaking the correct behaviour is to show the button as disabled.

 _But_ this, perhaps accidental, behaviour, facilitates my hacky way of having fields grouped in the cp, for layout/display purposes.

**Reply by mmikkel** [reply]
FWIW, I do think "Static Matrix Fields" as a concept has merit. I'd hope to see it implemented as a proper feature though, instead of having something that _sort of_ works similar to how a true static field would function, via juggling the types/min/max settings :)

**Comment by brandonkelly** [comment]
> I think I figured it out. You have to:
> 
> 1. Have `Min Entries` be equal to `Max Entries`
> 2. Have only one entry type allowed in the field settings.
> 3. Add the maximum blocks according to step 1.

Correct, that’s the intended behavior, and has been since 3.0.0 (86d4dbe4bf3567f6ff736d5f06f162587a9998e2).
